Applications/FileManager/main.cpp

```diff
@@ -104,6 +104,22 @@ int main(int argc, char** argv)
             if (rc < 0) {
                 GMessageBox::show(String::format("mkdir(\"%s\") failed: %s", new_dir_path.characters(), strerror(errno)), "Error", GMessageBox::Type::Error, GMessageBox::InputType::OK, window);
             } else {
+                file_system_model->update();
+
+                auto current_path = directory_view->path();
+
+                // not exactly sure why i have to reselect the root node first, but the index() fails if I dont
+                auto root_index = file_system_model->index(file_system_model->root_path());
+                tree_view->selection().set(root_index);
+                tree_view->scroll_into_view(root_index, Orientation::Vertical);
```

Comment:
Do we really need to `scroll_into_view` and `update` twice for the workaround as well?

Reply:
I'll play around with this tonight to see if everything is required or not.

Reply:
Looks as though just setting the index was required.